src/plugins/intel_gpu/src/graph/impls/ocl/multiclass_nms.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "multiclass_nms_inst.h"
+#include "primitive_base.hpp"
+#include "impls/implementation_map.hpp"
+#include "kernel_selector_helper.h"
+#include "multiclass_nms/multiclass_nms_kernel_ref.h"
+#include "multiclass_nms/multiclass_nms_kernel_selector.h"
+
+
+namespace cldnn {
+namespace ocl {
+struct multiclass_nms_impl : public typed_primitive_impl_ocl<multiclass_nms> {
+    using parent = typed_primitive_impl_ocl<multiclass_nms>;
+    using parent::parent;
+
+    std::unique_ptr<primitive_impl> clone() const override {
+        return make_unique<multiclass_nms_impl>(*this);
+    }
+
+protected:
+    kernel_arguments_data get_arguments(typed_primitive_inst<multiclass_nms>& instance, int32_t unused) const override {
+        kernel_arguments_data args = parent::get_arguments(instance, unused);
+        args.inputs.push_back(instance.output_indices_memory());
+        args.inputs.push_back(instance.output_num_memory());
+        return args;
+    }
+
+public:
+    static primitive_impl* create(const multiclass_nms_node& arg, const kernel_impl_params& impl_param) {
+        auto params = get_default_params<kernel_selector::multiclass_nms_params>(impl_param);
+        auto optional_params =
+            get_default_optional_params<kernel_selector::multiclass_nms_optional_params>(arg.get_program());
+
+        const auto& primitive = arg.get_primitive();
+
+        switch (primitive->sort_result) {
```

Comment:
could you extract at least switches to separate functions

Reply:
Done.